File: bookstore-backend/src/main/java/com/bookstore/bookstore/presentation/response/OrderItemResponse.java
package com.bookstore.bookstore.presentation.response;

import com.bookstore.bookstore.domain.enums.PurchaseItemType;
import java.math.BigDecimal;
import java.util.UUID;

public record OrderItemResponse(
        UUID id,
        PurchaseItemType itemType,
        UUID bookId,
        UUID digitalAssetId,
        String bookTitle,
        BigDecimal unitPrice,
        int quantity,
        BigDecimal lineTotal
) {
}

File: bookstore-desktop/Bookstore.Desktop/Dtos/OrderItemResponse.cs
namespace Bookstore.Desktop.Dtos;

public class OrderItemResponse
{
    public string Id { get; init; } = "";
    public string BookId { get; init; } = "";
    public string BookTitle { get; init; } = "";
    public decimal UnitPrice { get; init; }
    public int Quantity { get; init; }
    public decimal LineTotal { get; init; }
}
